File: src/main/kotlin/com/wtw/demo/comparator/UnsupportedDiffer.java
package com.wtw.demo.comparator;

import java.util.Objects;
import java.util.Optional;

public class UnsupportedDiffer implements Differ<Boolean> {

    @Override
    public <S extends Boolean> Optional<DiffResult> diff(S a, S b) {
        return Optional.of(DiffResult.error(a,b,"diff not supported"));
    }
}

File: src/main/kotlin/com/wtw/demo/comparator/UnsupportedDiffer.kt
package com.wtw.demo.comparator

import java.util.*

class UnsupportedDiffer : Differ<Boolean> {
    override fun diff(a: Boolean?, b: Boolean?): Optional<DiffResult> {
        return Optional.of(DiffResult.error(a, b, "diff not supported"))
    }
}
